topdown/strings.go

```diff
@@ -310,6 +310,37 @@ func builtinContains(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term
 	return iter(ast.BooleanTerm(strings.Contains(string(s), string(substr))))
 }
 
+func builtinStringCount(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+	s, err := builtins.StringOperand(operands[0].Value, 1)
+	if err != nil {
+		return err
+	}
+
+	substr, err := builtins.StringOperand(operands[1].Value, 2)
+	if err != nil {
+		return err
+	}
+
+	if len(substr) == 0 {
+		return iter(ast.IntNumberTerm(len(s) + 1))
+	}
+
+	baseTerm := string(s)
+	searchTerm := string(substr)
+
+	count := 0
+	for {
```

Comment:
I'd have expected `strings.Count` here. Is it a bad fit?

Reply:
Sorry, I thought the implementation has to be the complete algorithm, without using the library function. Happy to change it, if that's the preferred option.

Reply:
It's no problem per se -- however, looking at the implementation of [`strings.Count`](https://pkg.go.dev/strings#Count), the [source here](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/strings/strings.go;l=40), we find that it's equivalent to your algorithm [for this part](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/strings/strings.go;l=48-57), but has optimizations for the ["len(searchTeam) == 1" case here](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/strings/strings.go;l=45-47). I think that it's common enough to search for one character (":" and "/" come to mind) that it's worth prefering `strings.Count`. 😃

Reply:
@srenatus Updated accordingly